### PR TITLE
sys: add hwtimer compatibility layer for periph timers

### PR DESCRIPTION
**Pull request by kaspar030**
some platforms don't implement hwtimer_arch directly on timer hardware, but using one periph timer.
In order to reduce code duplication, factor that layer into a module in sys.

This PR also switches samd21 CPU to use this layer.

### CODE REVIEW

**Comment by kaspar030** [comment]
I don't like those compatibility layers, but very probably hwtimer will get dropped, anyways.

File: sys/compat/hwtimer/hwtimer_arch.c [hunk 1]
```diff
@@ -7,13 +7,13 @@
...
  * @{
  *
  * @file        hwtimer_arch.c
- * @brief       Implementation of the kernels hwtimer interface
+ * @brief       Implementation of the kernels hwtimer interface over periph timers
  *
- * The hardware timer implementation uses the Cortex build-in system timer as back-end.
+ * This hwtimer implementation wraps one periph timer 
  *
  * @author      Thomas Eichinger <thomas.eichinger@fu-berlin.de>
```
Comment (haukepetersen) [inline sys/compat/hwtimer/hwtimer_arch.c]: you could put your name in here, as you introduce this module...

**Comment by haukepetersen** [comment]
Looks good to me, it's up to you if you want to address my comment above... ACK

**Comment by kaspar030** [comment]
Well, I like my name, so I added it. It's still a copy of Thomas' file. :)

**Comment by haukepetersen** [comment]
No, actually I was the one to write it originally :-) re-ACK, merge when Travis is happy

**Comment by kaspar030** [comment]
- squashed, waiting for travis...

**Comment by kaspar030** [comment]
and GO.

**Pull request merged**